### PR TITLE
Documentation hierarchy

### PR DESCRIPTION
**Pull request by anarthal**
This PR attempts to create a structure for the guides so that we can keep writing content. It attempts to follow what was discussed yesterday with @alandefreitas and @PeterTurcan.

All the old content has been placed under the new hierarchy, or removed when sensible. The content **has not** been updated.

For the new topics, I've noted some suggestions. I don't know if this works for your way of managing this project - we can convert these into GitHub issues if you feel it's better.

The release process section has been merged into User and Contributor guides - pending content update to match these two user profiles.

### CODE REVIEW

File: contributor-guide/modules/ROOT/pages/becoming-author/docs-requirements.adoc
Comment (PeterTurcan) [inline contributor-guide/modules/ROOT/pages/becoming-author/docs-requirements.adoc:1]: Consider using Title Case for all headlines. 
Title Case is Capitalizing all Important Words
This is sentence case - capitalizing only the first word
This Is Capitalizing All Words

File: contributor-guide/modules/ROOT/pages/getting-involved.adoc
Comment (PeterTurcan) [inline contributor-guide/modules/ROOT/pages/getting-involved.adoc:1]: Title Case for headlines:  Getting Involved

File: user-guide/modules/ROOT/nav.adoc
Comment (PeterTurcan) [inline user-guide/modules/ROOT/nav.adoc:1]: I would be tempted to add a bit more structure to the user guide.

Introduction
What's in Boost
Getting Started
**Building Boost**
     - BSL
     - Installing Selected Libraries
     - Advanced Building
**Using Boost**
      - Combining Libraries
      - Testing and Debugging
      - FAQ
      - Reporting Bugs
**Tutorials**
          - How to ABC
          - How to DEF
**The Boost Community** (or Ecosystem or similar term)
      - Release Process
      - History of Boost
      etc.
**Reference**
      - Library Naming
      - Header Organization
      etc.

File: contributor-guide/modules/ROOT/pages/becoming-author/docs-requirements.adoc
Comment (vinniefalco) [reply contributor-guide/modules/ROOT/pages/becoming-author/docs-requirements.adoc:1]: OKAY BUT WHAT DO YOU CALL CAPITALIZING *ALL WORDS* AND ALSO BOLDING SOME OF THE TEXT FOR _EMPHASIS_ ?

Comment (PeterTurcan) [reply contributor-guide/modules/ROOT/pages/becoming-author/docs-requirements.adoc:1]: YELLING?

Comment (vinniefalco) [reply contributor-guide/modules/ROOT/pages/becoming-author/docs-requirements.adoc:1]: Fair, but what do you call it when you change the letter "i" to the letters "oi" and then make a cultural reference. For example:

"Oi, You Got a Loicense for that Toitle Case?"

File: contributor-guide/modules/ROOT/pages/testing/ci.adoc
[the remark this answers is not in the document]
Comment (vinniefalco) [reply contributor-guide/modules/ROOT/pages/testing/ci.adoc:1]: Hey these are good tips should we collect them into a style guide adoc file?

**Comment by vinniefalco** [comment]
Some of these toc headings need work. Some ideas:

"Becoming a library author" -> "Become an Author"
or
"Becoming a library author" -> "Author a Library"

"Source control" -> "Version Control (Git)"

"Library Design Guidelines" -> "Design Guide"

"Managing dependencies" -> "Dependencies"

"Boost Header policy" -> "Include Lines"

"Reporting bugs" -> "Reporting Issues"

Too much of the same word here:

image

**Comment by vinniefalco** [comment]
"Building Boost: advanced uses cases" -> "Advanced Builds"

"What’s in Boost?" -> "Use Cases"
or
"What’s in Boost?" -> "What's Inside"

Too much of the word Boost here:

image

File: user-guide/modules/ROOT/nav.adoc
Comment (anarthal) [reply user-guide/modules/ROOT/nav.adoc:1]: Cool. What's the reasoning behind putting the license under building boost?

Comment (PeterTurcan) [reply user-guide/modules/ROOT/nav.adoc:1]: @anarthal Not great reasoning, perhaps under Using Boost or even The Boost Community makes more sense.

File: user-guide/modules/ROOT/pages/whats-in-boost.adoc [hunk 1]
```diff
@@ -0,0 +1,13 @@
+= What's in Boost?
+
+@anarthal: TBC: content. Talk about the different useful stuff we have, for example:
```
Comment (alandefreitas) [inline user-guide/modules/ROOT/pages/whats-in-boost.adoc:3]: The content about libraries working together should move here. I don't even know where it is anymore.

In any case, I believe the initial motivation for this page for Vinnie was different from what we have now and we'll have to deal with that somehow with different content.

When Vinnie asked for libraries that work together, I think the main point was to sell the idea that boost libraries depending on each other is a strength not a weakness. This is valid point to communicate because there's some resistance to dependencies in C++ libraries.

Another point to communicate in some other content is really "What's in boost". The categories from "meta/libraries.json" are not relevant here because we need the perspective of the user here. One good use case is replacements for or extensions to the STL libraries, this can branch into replacements for performance reasons or because the user is stuck with an older compiler. We also have networking, which is huge. 

But we can't center the discussion around those two cases. Boost needs to be more than that. Two sources of ideas could be just checking boost libraries other boost libraries are using. Or just looking at the graph and clicking the libraries to see what are the common subgraphs users are prone to consume together. For instance, https://alandefreitas.github.io/boostdep_graph/libs/mysql.html.

Another idea is just taking use cases directly from other repositories and explaining how boost helps them. For instance, bitcoin-core is probably one of the largest project on github and it uses boost. Or at least used to use.

**Review by alandefreitas — APPROVED** [review]

File: user-guide/modules/ROOT/nav.adoc
Comment (alandefreitas) [reply user-guide/modules/ROOT/nav.adoc:1]: Yes. The nav bars are a little unbalanced sometimes.

**Comment by anarthal** [comment]
> Include Lines

Include Lines? in any case, this is old content, IMHO should be merged with design guide.

File: contributor-guide/modules/ROOT/nav.adoc [hunk 2]
```diff
@@ -1,23 +1,34 @@
...
-** xref:portability-requirements.adoc[]
-** xref:organization-requirements.adoc[]
-** xref:library-design-guidelines.adoc[]
-* xref:antora.adoc[Antora Guide]
-* Legacy Content
-** xref:documentation-overview.adoc[]
-** xref:documentation-structure-guidelines.adoc[]
+* xref:getting-involved.adoc[]
+* xref:becoming-author/intro.adoc[]
+** xref:becoming-author/license-requirements.adoc[]
```
[the remark this answers is not in the document]
Comment (anarthal) [reply contributor-guide/modules/ROOT/nav.adoc:4]: We probably want to coalesce the requirements sections. I didn't want to touch more than required, this is already big.

**Comment by alandefreitas** [comment]
> Include Lines? 

Yep. That's weird. 

@anarthal Is this good to go?

File: contributor-guide/modules/ROOT/nav.adoc [hunk 3]
```diff
@@ -1,23 +1,34 @@
 
-
```
[the remark this answers is not in the document]
Comment (vinniefalco) [reply contributor-guide/modules/ROOT/nav.adoc:2]: or how about what I said which is "Author a Library"

[the remark this answers is not in the document]
Comment (alandefreitas) [reply contributor-guide/modules/ROOT/nav.adoc:2]: Yes. "Author a Library" sounds better to me.

File: user-guide/modules/ROOT/pages/release-process.adoc [hunk 4]
```diff
@@ -0,0 +1,69 @@
+= The Boost Release Process
```
[the remark this answers is not in the document]
Comment (vinniefalco) [reply user-guide/modules/ROOT/pages/release-process.adoc:1]: Stop saying Boost on every page. "Release Process" is sufficient

File: contributor-guide/modules/ROOT/pages/testing/writing-tests.adoc [hunk 5]
```diff
@@ -0,0 +1,11 @@
+= Writing tests
+
+@anarthal: content suggestion:
+
+* Using a testing framework vs. executables
+* The framework we can use: Boost.Test and lightweight test. What we can't use and why (gtest, catch2)
+* Running tests from b2 and cmake
+* Running examples as tests
+* Running docs snippets as tests
+* What if I have integration tests?
```
[the remark this answers is not in the document]
Comment (anarthal) [reply contributor-guide/modules/ROOT/pages/testing/writing-tests.adoc:10]: how many libraries in Boost have integration tests? I think there's a total of 2

[the remark this answers is not in the document]
Comment (anarthal) [reply contributor-guide/modules/ROOT/pages/testing/writing-tests.adoc:10]: (Integration tests, in this context, meaning "something that requires an external system", such as a Redis server or a MySQL instance)

File: contributor-guide/modules/ROOT/pages/becoming-author/intro.adoc [hunk 6]
```diff
@@ -1,5 +1,4 @@
-= Introduction to becoming a Boost Contributor
-:navtitle: Becoming a Boost Contributor
```
[the remark this answers is not in the document]
Comment (vinniefalco) [reply contributor-guide/modules/ROOT/pages/becoming-author/intro.adoc:2]: "Becoming a Boost Contributor" is too wordy.

"Becoming a Boost Contributor" -> "Contributing"

File: user-guide/modules/ROOT/nav.adoc
Comment (anarthal) [reply user-guide/modules/ROOT/nav.adoc:1]: Applied everything except for the building Boost/using Boost groupings, since that depends on splitting the getting started for different systems, which IIRC @alandefreitas has done in a different PR.

File: contributor-guide/modules/ROOT/nav.adoc [hunk 7]
```diff
@@ -1,23 +1,34 @@
...
+** xref:becoming-author/docs-requirements.adoc[]
+** xref:becoming-author/library-design-guidelines.adoc[]
+** xref:becoming-author/managing-dependencies.adoc[]
+** xref:becoming-author/separate-compilation.adoc[]
+** xref:becoming-author/header.adoc[]
+** xref:becoming-author/borland.adoc[]
+** xref:becoming-author/library-metadata.adoc[]
+** xref:becoming-author/release-notes.adoc[]
+* xref:source-control.adoc[]
+* xref:cmake.adoc[]
```
[the remark this answers is not in the document]
Comment (anarthal) [reply contributor-guide/modules/ROOT/nav.adoc:16]: We probably need a page on cmake in the user guide too, then. I'll make a mention in the advanced building Boost, and change this comment. I'll add a page for b2 here too.

File: contributor-guide/modules/ROOT/pages/testing/writing-tests.adoc [hunk 8]
```diff
@@ -0,0 +1,11 @@
+= Writing tests
+
+@anarthal: content suggestion:
```
[the remark this answers is not in the document]
Comment (anarthal) [reply contributor-guide/modules/ROOT/pages/testing/writing-tests.adoc:3]: Maintaining as per your later comment.

File: contributor-guide/modules/ROOT/pages/testing/writing-tests.adoc (hunk 5 above)
[the remark this answers is not in the document]
Comment (anarthal) [reply contributor-guide/modules/ROOT/pages/testing/writing-tests.adoc:10]: Well these are not intended as section names per-se, but as suggestions about things I'd have liked to read when I became a Boost author. I think tailoring these should be part of subsequent stories.

[the remark this answers is not in the document]
Comment (vinniefalco) [reply contributor-guide/modules/ROOT/pages/testing/writing-tests.adoc:10]: image

File: user-guide/modules/ROOT/pages/whats-in-boost.adoc (hunk 1 above)
Comment (anarthal) [reply user-guide/modules/ROOT/pages/whats-in-boost.adoc:3]: I've noted your ideas in the file

File: contributor-guide/modules/ROOT/nav.adoc (hunk 3 above)
[the remark this answers is not in the document]
Comment (anarthal) [reply contributor-guide/modules/ROOT/nav.adoc:2]: Went to "Author a Library", which looks great but the sub-categories are still too many. I've ended up splitting this topic into "requirements" and "design guide"

File: contributor-guide/modules/ROOT/pages/testing/writing-tests.adoc (hunk 5 above)
[the remark this answers is not in the document]
Comment (alandefreitas) [reply contributor-guide/modules/ROOT/pages/testing/writing-tests.adoc]: > how many libraries in Boost have integration tests? I think there's a total of 2

We have these tests on our libraries. If you use the boost-ci template, you'll get them.

[the remark this answers is not in the document]
Comment (alandefreitas) [reply contributor-guide/modules/ROOT/pages/testing/writing-tests.adoc]: > meaning "something that requires an external system",

Oh... then no.
